### PR TITLE
Extract object reflection management from Context.

### PR DESCRIPTION
**Pull request by eugeneko**
@rokups @gleblebedev I touched C# more than ever in this PR, you may want to check it.

### CODE REVIEW

File: Source/Urho3D/CSharp/Managed/Core/Context.cs [hunk 1]
```diff
@@ -106,6 +106,11 @@ public void RegisterFactory(Type type, string category="")
                 throw new InvalidOperationException("This type can not be registered as factory.");
 
             Urho3D_Context_RegisterFactory(getCPtr(this), type.Name, StringHash.Calculate("SwigDirector_" + baseType.Name), category);
+            if (type.IsSubclassOf(typeof(Serializable)))
+            {
+                var serializable = (Serializable)Activator.CreateInstance(type, new object[] { Context.Instance });
+                serializable.RegisterAttributes();
```
Comment (rokups) [inline Source/Urho3D/CSharp/Managed/Core/Context.cs:112]: Not strictly incorrect, but better would be:
```cs
using (var serializable = (Serializable)Activator.CreateInstance(type, new object[] { Context.Instance }))
{
    serializable.RegisterAttributes();
}
```
This essentially calls `serializable.Dispose()` automatically and thus gets rid of an object immediately.

File: Source/Urho3D/CSharp/Native/Context.cpp [hunk 2]
```diff
@@ -39,48 +39,21 @@ extern Urho3D_CSharpCreateObjectCallback Urho3D_CSharpCreateObject;
...
-
-protected:
-    /// Hash of base type (SWIG director class).
-    StringHash baseType_;
-    /// Hash of managed type. This is different from `baseType_`.
-    StringHash managedType_;
-};
+    const StringHash managedType = typeInfo->GetType();
+    auto object = reinterpret_cast<Object*>(Urho3D_CSharpCreateObject(context, managedType.Value()));
+    return SharedPtr<Object>(object);
```
Comment (rokups) [inline Source/Urho3D/CSharp/Native/Context.cpp:46]: I love how much simpler this got 😍

**Comment by rokups** [comment]
I love how much simpler some parts got 👍🏻 There is a lot of stuff removed from `CameraViewport`. Whats that about? No longer compatible with new renderer?

**Comment by eugeneko** [comment]
CameraViewport was about legacy render path and post-processing, and I don't see any reason to spend even a minute refactoring this code, would be wasted effort.

**Comment by eugeneko** [comment]
Do you think it's okay to merge?
I made these changes because I wanted _quick_ attribute search via cache-friendly array, and adding anything to the Context was pain the the back

**Comment by rokups** [comment]
Sure why not. Looks like only good things here

**Pull request merged**